### PR TITLE
Add a denorm test fix

### PR DESCRIPTION
**Pull request by geyyer**
Adjust data type for conv_fwd, conv_bwd_data and conv_bwd_weight fp16 -> bf16, add type_convert implementations.

### CODE REVIEW

**Comment by zjing14** [comment]
@asroy should we hold this PR til the performance issue of the f16->bf16 conversion is resolved?

File: include/ck/tensor_operation/gpu/grid/gridwise_gemm_multiple_d_xdl_cshuffle.hpp [hunk 1]
```diff
@@ -92,6 +92,13 @@ struct GridwiseGemmMultipleD_xdl_cshuffle
     using GridwiseGemmPipe = remove_cvref_t<decltype(
         GridwiseGemmPipeline_Selector<PipelineVer, NumGemmKPrefetchStage, LoopSched>())>;
 
+    // denorm test fix, required to work around fp16 mfma issue
+    // we convert fp16->fp32->bf16 and execute bf16 mfma instruction
+    // when mfma if fixed, remove this section and update
+    // ABDataTypeAdjusted -> ABDataType throughout this file
+    using ABDataTypeAdjusted =
```
Comment (zjing14) [inline include/ck/tensor_operation/gpu/grid/gridwise_gemm_multiple_d_xdl_cshuffle.hpp:99]: We can have a macro here to turn on the workaround on MI200 only. Like
`
#if defined(__gfx90a__)
 using ABDataTypeAdjusted = conditional_t<is_same_v<ABDataType, ck::half_t>, ck::bhalf_t, ABDataType>;
#esle
 using ABDataTypeAdjusted = ABDataType;
#endif 
`

Comment (geyyer) [reply include/ck/tensor_operation/gpu/grid/gridwise_gemm_multiple_d_xdl_cshuffle.hpp:100]: Done in [3360a1f](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/603/commits/3360a1f5c36af0a8d3e88afc5bf7776d762fee75)

**Review by zjing14 — APPROVED** [review]

**Comment by atafra** [comment]
Replacing FP16 MMA with BF16 does not work for all workloads, the precision is far too low. See: https://github.com/ROCmSoftwarePlatform/composable_kernel/issues/671

**Pull request merged**